### PR TITLE
Add tests for validating schema files.

### PR DESCRIPTION
**Pull request by xuhdev**
Run lint programs and try to use the latest PyDAX to load each dataset
once.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,3 +2,13 @@
 
 This is the repository that we maintain the default DAX schema files for
 [PyDAX](https://github.com/codait/pydax).
+
+To run tests locally, run:
+
+    $ pip install -U -rrequirements/tox.txt  # If you are inside a virtual environment, conda environment
```
Comment (djalova) [inline README.md:8]: Extra r in requirements

Comment (xuhdev) [reply README.md]: There's a missing space between `-r` and `requirements`. Added

File: tests/test_pydax.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+import pydax
+
+
+pydax.init(
+    DEFAULT_DATASET_SCHEMA_URL='https://raw.githubusercontent.com/CODAIT/dax-schemata/master/datasets.yaml',
+    DEFAULT_FORMAT_SCHEMA_URL='https://raw.githubusercontent.com/CODAIT/dax-schemata/master/formats.yaml',
+    DEFAULT_LICENSE_SCHEMA_URL='https://raw.githubusercontent.com/CODAIT/dax-schemata/master/licenses.yaml'
+)
+
+for name, versions in pydax.list_all_datasets().items():
```
Comment (bdwyer2) [inline tests/test_pydax.py:30]: Maybe we should test to make sure that `pydax.list_all_datasets()` returns the expected number of datasets? Or at least it returns more than 0?

File: .github/workflows/pydax.yml [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+# limitations under the License.
+#
+
+name: PyDAX
+
+on:  # yamllint disable-line rule:truthy
+  push:
+    branches: 'master'
+  pull_request:
+    branches: '*'
```
Comment (bdwyer2) [inline .github/workflows/pydax.yml:23]: Is there any benefit to testing PRs? Won't `pydax` be looking at the master branch?

File: tox.ini [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+[tox]
+envlist = lint, pydax
+skipsdist = true
+
+[testenv]
+download = True
+
+[testenv:lint]
+deps = -rrequirements/lint.txt  # We put them in this file so dependabot can discover it
```
Comment (bdwyer2) [inline tox.ini:25]:
```suggestion
deps = -r requirements/lint.txt  # We put them in this file so dependabot can discover it
```

File: tox.ini [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+commands =
+    yamllint -c .yamllint.yaml .
+    flake8 .
+
+[testenv:pydax]
+# We only let the latest development version of pydax to run these tests for now. After the first
+# release, we also need to test with the stable versions.
+deps = -rrequirements/pydax.txt
```
Comment (bdwyer2) [inline tox.ini:33]:
```suggestion
deps = -r requirements/pydax.txt
```

File: .github/workflows/pydax.yml (hunk 3 above)
Comment (xuhdev) [reply .github/workflows/pydax.yml:23]: The PRs update the data files and this would test whether the changes can pass minimum quality as input to PyDAX

Comment (bdwyer2) [reply .github/workflows/pydax.yml:23]: I guess what I'm saying is the PR would update this file (until it's merged):
https://raw.githubusercontent.com/CODAIT/dax-schemata/PR-BRANCH-NAME/datasets.yaml

And pydax would be downloading this:
https://raw.githubusercontent.com/CODAIT/dax-schemata/master/datasets.yaml

File: tests/test_pydax.py (hunk 2 above)
Comment (xuhdev) [reply tests/test_pydax.py:39]: Good point, added.

File: .github/workflows/pydax.yml (hunk 3 above)
Comment (xuhdev) [reply .github/workflows/pydax.yml:23]: Oh I got what you mean. The script need to be updated.

File: tests/test_pydax.py [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+
+import yaml
+
+import pydax
+
+
+pydax.init(
+    DEFAULT_DATASET_SCHEMA_URL='datasets.yaml',
+    DEFAULT_FORMAT_SCHEMA_URL='formats.yaml',
+    DEFAULT_LICENSE_SCHEMA_URL='licenses.yaml'
```
Comment (xuhdev) [inline tests/test_pydax.py:29]: @bdwyer2 Updated

File: tests/test_pydax.py [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+)
+
+with open('datasets.yaml') as f:
+    datasets = yaml.safe_load(f)
+
+# Datasets name are the same from the schema files. This helps ensure that PyDAX doesn't miss any dataset during the
+# test.
+assert frozenset(datasets['datasets']) == frozenset(pydax.list_all_datasets())
```
Comment (bdwyer2) [inline tests/test_pydax.py:37]: Do you think it makes sense to add a sanity check to make sure the number is > 0? I'm just worried about a small formatting error breaking this and going unnoticed.

If you don't think this is needed feel free to just close this comment.

Comment (xuhdev) [reply tests/test_pydax.py:37]: Makes sense, added

**Comment by xuhdev** [comment]
@bdwyer2 The PR should be ready for review again (the re-request review button is unclickable for some reason)

**Review by bdwyer2 — APPROVED** [review]

**Review by djalova — APPROVED** [review]

**Pull request merged**